### DISCUSSION TITLE
Problem getting key released event with USB HID Keyboard Host

### DISCUSSION THREAD

**Discussion by DatanoiseTV** [hathach/tinyusb#1739] [General]
I am implementing a USB Keyboard Host and I can successfully receive events when a key (or more, up to 6) get pressed, but I am not getting an event if a key is released.

```cpp
static void process_kbd_report(uint8_t dev_addr, hid_keyboard_report_t const *report)
{
    (void)dev_addr;
    static hid_keyboard_report_t prev_report = {0, 0, {0}}; // previous report to check key released
    bool flush = false;

    for (uint8_t i = 0; i < 6; i++)
    {
        uint8_t keycode = report->keycode[i];
        if (keycode)
        {
            keycode2midinote(keycode, report->modifier & (KEYBOARD_MODIFIER_LEFTSHIFT | KEYBOARD_MODIFIER_RIGHTSHIFT));
        }
    }
    prev_report = *report;
}
```

**Comment by hathach** [comment]
- can you specify your hw set up and which firmware you are running. 
- Did the stock example work for you ?

**Reply by DatanoiseTV** [reply]
The example from "examples/host/cdc_msc_hid/src/hid_app.c" reports also only keys being pressed down, but no release events.

**Reply by hathach** [reply]
> can you specify your hw set up and which firmware you are running.

which hardware setup you are using: board, mcu ?

**Reply by DatanoiseTV** [reply]
Raspberry Pico with RP2040

**Reply by hathach** [reply]
last time I tested with rp2040, it works just fine. 
1. Why do you think the report is not received ? 
2. Could you **attached debug log with LOG=2 as txt** (or paste it if it is short enough) and put some notes where you think the report should be received but not.
